### PR TITLE
added type hints

### PR DESCRIPTION
**Pull request by ghost**
Since the minimum supported version py3.8 supports type hints, type hints should be included where possible.
All tests succesfully for py3.8-py3.11
PR to master as no up to date dev branch is available.

### CODE REVIEW

File: tests/testdeb_0.0.0.dsc [hunk 1]
```diff
@@ -18,4 +18,3 @@ Checksums-Sha256:
 Files:
  142ca7334ed1f70302b4504566e0c233 280 testdeb_0.0.0.orig.tar.gz
  fc80e6e7f1c1a08b78a674aaee6c1548 232 testdeb_0.0.0-1.debian.tar.xz
-
```
Comment (memory) [inline tests/testdeb_0.0.0.dsc]: please undo this change, which will allow you to undo the changes to the checksums in test_dsc.py

**Review by memory — CHANGES_REQUESTED** [review]
Thank you so much for this; it's long overdue and well appreciated.

Two requests:

1. Please revert the removal of the final EOL from testdeb_0.0.0.dsc, which in turn will allow you to revert the checksum changes in test_dsc.py
2. Please add a mypy stage to the test harness (.github/workflows/pr.yaml) and ensure it passes

**Comment by ghost** [comment]
requested changes should be implemented.

**Review by memory — APPROVED** [review]
Thank you again for this!
